Agent&Driver cannot start in Spring Boot Fatjar packaging mode

## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
master

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
Agent + JDBC + Spring Boot(Fatjar)
 
### Expected behavior
Can start
### Actual behavior
Unable to start

```shell
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:491)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:503)
Caused by: java.lang.NoClassDefFoundError: org/apache/shardingsphere/infra/config/props/ConfigurationPropertyKey
	at org.apache.shardingsphere.agent.core.ShardingSphereAgent.premain(ShardingSphereAgent.java:57)
	... 6 more
Caused by: java.lang.ClassNotFoundException: org.apache.shardingsphere.infra.config.props.ConfigurationPropertyKey
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 7 more
*** java.lang.instrument ASSERTION FAILED ***: "result" with message agent load/premain call failed at open/src/java.instrument/share/native/libinstrument/JPLISAgent.c line: 422
FATAL ERROR in native method: processing of -javaagent failed, processJavaStart failed
[1]    76954 abort      java  -jar spring-boot-shardingsphere-jdbc-test.jar
```

```shell
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:491)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:503)
Caused by: java.lang.NoClassDefFoundError: org/apache/shardingsphere/infra/config/props/ConfigurationPropertyKey
	at org.apache.shardingsphere.agent.core.ShardingSphereAgent.premain(ShardingSphereAgent.java:57)
	... 6 more
Caused by: java.lang.ClassNotFoundException: org.apache.shardingsphere.infra.config.props.ConfigurationPropertyKey
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 7 more
*** java.lang.instrument ASSERTION FAILED ***: "result" with message agent load/premain call failed at open/src/java.instrument/share/native/libinstrument/JPLISAgent.c line: 422
FATAL ERROR in native method: processing of -javaagent failed, processJavaStart failed
[1]    81619 abort      java  -jar spring-boot-shardingsphere-jdbc-test.jar
```

### Reason analyze (If you can)

After Spring Boot is packaged, a custom class loader will load the class and start it. The corresponding dependencies cannot be relied on in the Agent.

#### Task
- [ ] Agent Core uses jdk native log output instead of slf4j and logback.
- [ ] Remove the dependency on shardingsphere in Agent Core, and move the judgment of whether to enable the plugin to each advice.


### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

### Example codes for reproduce this issue (such as a github link).

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/advisor/config/AdvisorConfigurationLoader.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/advisor/config/AdvisorConfigurationLoader.java
@@ -29,15 +29,12 @@ import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.jar.JarFile;
-import java.util.logging.Logger;
 
 /**
  * Advisor configuration loader.
  */
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public final class AdvisorConfigurationLoader {
-    
-    private static final Logger LOGGER = Logger.getLogger(AdvisorConfigurationLoader.class.getName());
     
     /**
      * Load advisor configurations.
@@ -51,9 +48,7 @@ public final class AdvisorConfigurationLoader {
         AgentPluginClassLoader agentPluginClassLoader = new AgentPluginClassLoader(Thread.currentThread().getContextClassLoader(), pluginJars);
         for (String each : pluginTypes) {
             InputStream advisorsResourceStream = getResourceStream(agentPluginClassLoader, each);
-            if (null == advisorsResourceStream) {
-                LOGGER.info(String.format("No configuration of advisor for type `%s`.", each));
-            } else {
+            if (null != advisorsResourceStream) {
                 mergeConfigurations(result, YamlAdvisorsConfigurationSwapper.swap(YamlAdvisorsConfigurationLoader.load(advisorsResourceStream), each));
             }
         }

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/advisor/executor/type/ConstructorAdviceExecutor.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/advisor/executor/type/ConstructorAdviceExecutor.java
@@ -34,6 +34,7 @@ import org.apache.shardingsphere.agent.core.advisor.executor.AdviceExecutor;
 import java.util.Collection;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
@@ -65,7 +66,7 @@ public final class ConstructorAdviceExecutor implements AdviceExecutor {
             // CHECKSTYLE:OFF
         } catch (final Throwable ex) {
             // CHECKSTYLE:ON
-            LOGGER.severe(String.format("Constructor advice execution error. class: %s, %s", target.getClass().getTypeName(), ex.getMessage()));
+            LOGGER.log(Level.SEVERE, "Constructor advice execution error. class: {0}, {1}", new String[]{target.getClass().getTypeName(), ex.getMessage()});
         }
     }
     

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/advisor/executor/type/InstanceMethodAdviceExecutor.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/advisor/executor/type/InstanceMethodAdviceExecutor.java
@@ -38,6 +38,7 @@ import java.util.Collection;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.concurrent.Callable;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
@@ -89,7 +90,7 @@ public final class InstanceMethodAdviceExecutor implements AdviceExecutor {
             // CHECKSTYLE:OFF
         } catch (final Throwable ex) {
             // CHECKSTYLE:ON
-            LOGGER.severe(String.format("Failed to execute the pre-method of method `%s` in class `%s`, %s.", method.getName(), target.getClass(), ex.getMessage()));
+            LOGGER.log(Level.SEVERE, "Failed to execute the pre-method of method `{0}` in class `{1}`, {2}.", new String[]{method.getName(), target.getClass().getName(), ex.getMessage()});
         }
     }
     
@@ -105,7 +106,7 @@ public final class InstanceMethodAdviceExecutor implements AdviceExecutor {
             // CHECKSTYLE:OFF
         } catch (final Throwable ignored) {
             // CHECKSTYLE:ON
-            LOGGER.severe(String.format("Failed to execute the error handler of method `%s` in class `%s`, %s.", method.getName(), target.getClass(), ex.getMessage()));
+            LOGGER.log(Level.SEVERE, "Failed to execute the error handler of method `{0}` in class `{1}`, {2}.", new String[]{method.getName(), target.getClass().getName(), ex.getMessage()});
         }
     }
     
@@ -121,7 +122,7 @@ public final class InstanceMethodAdviceExecutor implements AdviceExecutor {
             // CHECKSTYLE:OFF
         } catch (final Throwable ex) {
             // CHECKSTYLE:ON
-            LOGGER.severe(String.format("Failed to execute the post-method of method `%s` in class `%s`, %s.", method.getName(), target.getClass(), ex.getMessage()));
+            LOGGER.log(Level.SEVERE, "Failed to execute the post-method of method `{0}` in class `{1}`, {2}.", new String[]{method.getName(), target.getClass().getName(), ex.getMessage()});
         }
     }
     

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/advisor/executor/type/StaticMethodAdviceExecutor.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/advisor/executor/type/StaticMethodAdviceExecutor.java
@@ -36,6 +36,7 @@ import java.util.Collection;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.concurrent.Callable;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
@@ -87,7 +88,7 @@ public final class StaticMethodAdviceExecutor implements AdviceExecutor {
             // CHECKSTYLE:OFF
         } catch (final Throwable ex) {
             // CHECKSTYLE:ON
-            LOGGER.severe(String.format("Failed to execute the pre-method of method `%s` in class `%s`, %s.", method.getName(), klass, ex.getMessage()));
+            LOGGER.log(Level.SEVERE, "Failed to execute the pre-method of method `{0}` in class `{1}`, {2}.", new String[]{method.getName(), klass.getClass().getName(), ex.getMessage()});
         }
     }
     
@@ -103,7 +104,7 @@ public final class StaticMethodAdviceExecutor implements AdviceExecutor {
             // CHECKSTYLE:OFF
         } catch (final Throwable ignored) {
             // CHECKSTYLE:ON
-            LOGGER.severe(String.format("Failed to execute the error handler of method `%s` in class `%s`, %s.", method.getName(), klass, ex.getMessage()));
+            LOGGER.log(Level.SEVERE, "Failed to execute the error handler of method `{0}` in class `{1}`, {2}.", new String[]{method.getName(), klass.getClass().getName(), ex.getMessage()});
         }
     }
     
@@ -119,7 +120,7 @@ public final class StaticMethodAdviceExecutor implements AdviceExecutor {
             // CHECKSTYLE:OFF
         } catch (final Throwable ex) {
             // CHECKSTYLE:ON
-            LOGGER.severe(String.format("Failed to execute the post-method of method `%s` in class `%s` %s.", method.getName(), klass, ex.getMessage()));
+            LOGGER.log(Level.SEVERE, "Failed to execute the post-method of method `{0}` in class `{1}` {2}.", new String[]{method.getName(), klass.getClass().getName(), ex.getMessage()});
         }
     }
     

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/AgentTransformer.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/AgentTransformer.java
@@ -26,7 +26,6 @@ import net.bytebuddy.pool.TypePool.Default;
 import net.bytebuddy.utility.JavaModule;
 import org.apache.shardingsphere.agent.api.PluginConfiguration;
 import org.apache.shardingsphere.agent.core.advisor.config.AdvisorConfiguration;
-import org.apache.shardingsphere.agent.core.advisor.config.AdvisorConfigurationLoader;
 import org.apache.shardingsphere.agent.core.advisor.config.MethodAdvisorConfiguration;
 import org.apache.shardingsphere.agent.core.builder.interceptor.AgentBuilderInterceptChainEngine;
 import org.apache.shardingsphere.agent.core.builder.interceptor.impl.MethodAdvisorBuilderInterceptor;
@@ -48,7 +47,7 @@ import java.util.logging.Logger;
 @RequiredArgsConstructor
 public final class AgentTransformer implements Transformer {
     
-    private static final Logger LOGGER = Logger.getLogger(AdvisorConfigurationLoader.class.getName());
+    private static final Logger LOGGER = Logger.getLogger(AgentTransformer.class.getName());
     
     private static final Map<AgentPluginClassLoader, TypePool> TYPE_POOL_MAP = new ConcurrentHashMap<>();
     

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/AgentTransformer.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/AgentTransformer.java
@@ -39,6 +39,7 @@ import java.util.Collection;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.jar.JarFile;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
@@ -78,7 +79,7 @@ public final class AgentTransformer implements Transformer {
                 result.getAdvisors().add(each);
                 continue;
             }
-            LOGGER.severe(String.format("The advice class `%s` does not exist", each.getAdviceClassName()));
+            LOGGER.log(Level.SEVERE, "The advice class `{0}` does not exist", new String[]{each.getAdviceClassName()});
         }
         return result;
     }

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/interceptor/impl/MethodAdvisorBuilderInterceptor.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/interceptor/impl/MethodAdvisorBuilderInterceptor.java
@@ -27,6 +27,7 @@ import org.apache.shardingsphere.agent.core.builder.interceptor.AgentBuilderInte
 import org.apache.shardingsphere.agent.core.plugin.classloader.ClassLoaderContext;
 
 import java.util.Optional;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
@@ -58,7 +59,7 @@ public final class MethodAdvisorBuilderInterceptor implements AgentBuilderInterc
                 // CHECKSTYLE:OFF
             } catch (final Throwable ex) {
                 // CHECKSTYLE:ON
-                LOGGER.severe(String.format("Failed to load advice class: %s, %s.", typePointcut.getTypeName(), ex.getMessage()));
+                LOGGER.log(Level.SEVERE, "Failed to load advice class: {0}, {1}", new String[]{typePointcut.getTypeName(), ex.getMessage()});
             }
         }
         return result;

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/plugin/PluginLifecycleServiceManager.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/plugin/PluginLifecycleServiceManager.java
@@ -29,6 +29,7 @@ import java.util.Map;
 import java.util.Map.Entry;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.jar.JarFile;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
@@ -71,12 +72,12 @@ public final class PluginLifecycleServiceManager {
     
     private static void start(final PluginConfiguration pluginConfig, final PluginLifecycleService pluginLifecycleService, final boolean isEnhancedForProxy) {
         try {
-            LOGGER.info(String.format("Start plugin: %s", pluginLifecycleService.getType()));
+            LOGGER.log(Level.INFO, "Start plugin: {0}", new String[]{pluginLifecycleService.getType()});
             pluginLifecycleService.start(pluginConfig, isEnhancedForProxy);
             // CHECKSTYLE:OFF
         } catch (final Throwable ex) {
             // CHECKSTYLE:ON
-            LOGGER.severe(String.format("Failed to start service %s.", ex.getMessage()));
+            LOGGER.log(Level.SEVERE, "Failed to start service {0}.", new String[]{ex.getMessage()});
         }
     }
     
@@ -86,7 +87,7 @@ public final class PluginLifecycleServiceManager {
             try {
                 each.close();
             } catch (final IOException ex) {
-                LOGGER.severe(String.format("Failed to close jar file %s.", ex.getMessage()));
+                LOGGER.log(Level.SEVERE, "Failed to close jar file {0}.", new String[]{ex.getMessage()});
             }
         });
     }

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/plugin/jar/PluginJarLoader.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/plugin/jar/PluginJarLoader.java
@@ -31,6 +31,7 @@ import java.nio.file.attribute.BasicFileAttributes;
 import java.util.Collection;
 import java.util.LinkedList;
 import java.util.jar.JarFile;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
@@ -53,7 +54,7 @@ public final class PluginJarLoader {
         Collection<JarFile> result = new LinkedList<>();
         for (File each : jarFiles) {
             result.add(new JarFile(each, true));
-            LOGGER.info(String.format("Loaded jar: %s", each.getName()));
+            LOGGER.log(Level.INFO, "Loaded jar: {0}", new String[]{each.getName()});
         }
         return result;
     }

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/spi/AgentServiceLoader.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/spi/AgentServiceLoader.java
@@ -39,13 +39,8 @@ public final class AgentServiceLoader<T> {
     private final Collection<T> services;
     
     private AgentServiceLoader(final Class<T> service) {
-        validate(service);
-        this.services = load(service);
-    }
-    
-    private void validate(final Class<T> service) {
-        AgentPreconditions.checkNotNull(service, "SPI class is null.");
         AgentPreconditions.checkArgument(service.isInterface(), String.format("SPI class `%s` is not interface.", service));
+        this.services = load(service);
     }
     
     private Collection<T> load(final Class<T> service) {

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/util/AgentPreconditions.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/util/AgentPreconditions.java
@@ -40,19 +40,6 @@ public final class AgentPreconditions {
     }
     
     /**
-     * Check not null.
-     *
-     * @param reference reference
-     * @param errorMessage error message
-     * @throws NullPointerException null pointer exception
-     */
-    public static void checkNotNull(final Object reference, final String errorMessage) {
-        if (null == reference) {
-            throw new NullPointerException(errorMessage);
-        }
-    }
-    
-    /**
      * Check argument.
      *
      * @param condition condition

--- a/agent/plugins/core/src/main/java/org/apache/shardingsphere/agent/plugin/core/spi/PluginServiceLoader.java
+++ b/agent/plugins/core/src/main/java/org/apache/shardingsphere/agent/plugin/core/spi/PluginServiceLoader.java
@@ -37,13 +37,8 @@ public final class PluginServiceLoader<T> {
     private final Collection<T> services;
     
     private PluginServiceLoader(final Class<T> service) {
-        validate(service);
-        this.services = load(service);
-    }
-    
-    private void validate(final Class<T> service) {
-        PluginPreconditions.checkNotNull(service, "SPI class is null.");
         PluginPreconditions.checkArgument(service.isInterface(), String.format("SPI class `%s` is not interface.", service));
+        this.services = load(service);
     }
     
     private Collection<T> load(final Class<T> service) {

--- a/agent/plugins/core/src/main/java/org/apache/shardingsphere/agent/plugin/core/util/PluginPreconditions.java
+++ b/agent/plugins/core/src/main/java/org/apache/shardingsphere/agent/plugin/core/util/PluginPreconditions.java
@@ -27,32 +27,6 @@ import lombok.NoArgsConstructor;
 public final class PluginPreconditions {
     
     /**
-     * Check state.
-     *
-     * @param state state
-     * @param errorMessage error message
-     * @throws IllegalStateException illegal state exception
-     */
-    public static void checkState(final boolean state, final String errorMessage) {
-        if (!state) {
-            throw new IllegalStateException(errorMessage);
-        }
-    }
-    
-    /**
-     * Check not null.
-     *
-     * @param reference reference
-     * @param errorMessage error message
-     * @throws NullPointerException null pointer exception
-     */
-    public static void checkNotNull(final Object reference, final String errorMessage) {
-        if (null == reference) {
-            throw new NullPointerException(errorMessage);
-        }
-    }
-    
-    /**
      * Check argument.
      *
      * @param condition condition

